server/src/main/java/com/paypal/selion/proxy/SeLionRemoteProxy.java

```diff
@@ -166,10 +166,7 @@ public boolean release(String downloadJSON) {
 
         synchronized (this) {
             forceShutDown = true;
-            shutDownInProgress = true;
-            if (getTotalUsed() <= 1) {
-                shutdownNode();
-            }
+            startNodeRecycleThread();
```

Comment:
Minor but isn't there a possibility of two threads starting here on release upgrade cycle?